python/external_config/external_config_loader.py

```diff
@@ -39,7 +41,7 @@ class ExternalConfigurationLoader(QtCore.QObject):
 
     # signal emitted to indicate that an update has been detected
     # to the pipeline configurations for a project
-    configurations_loaded = QtCore.Signal(int, list)  # project_id, list of configs
+    configurations_loaded = QtCore.Signal(int, list, tuple)  # project_id, list of configs, error
```

Comment:
would it make sense to have a separate signal to indicate failure?

Also thinking if we rather than passing a string from this library should pass an error state/enum or something and then let desktop turn that into a response for the UI (in our case a string). Thinking about localization but also thinking this might give the caller a lot more flexibility in how errors are communicated to the user. With the string, all that the client code knows is that something went wrong. We already have a  `ExternalConfigNotAccessibleError` exception type bubbling up, so we got a good idea of the type of error.